pkg/kubelet/auth.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"net/http"
+
+	"k8s.io/kubernetes/pkg/auth/authenticator"
+	"k8s.io/kubernetes/pkg/auth/authorizer"
+	"k8s.io/kubernetes/pkg/auth/user"
+)
+
+// KubeletAuth implements AuthInterface
+type KubeletAuth struct {
+	// authenticator identifies the user for requests to the Kubelet API
+	// if unspecified, an empty user.Info is provided
+	authenticator authenticator.Request
+	// authorizerAttributeGetter builds authorization.Attributes for a request to the Kubelet API
+	// if unspecified, an empty attributes record is provided
+	authorizerAttributeGetter authorizer.RequestAttributesGetter
+	// authorizer determines whether a given authorization.Attributes is allowed
+	// if unspecified, all requests are allowed
+	authorizer authorizer.Authorizer
+}
+
+func NewKubeletAuth(authenticator authenticator.Request, authorizerAttributeGetter authorizer.RequestAttributesGetter, authorizer authorizer.Authorizer) *KubeletAuth {
+	return &KubeletAuth{
+		authenticator:             authenticator,
+		authorizerAttributeGetter: authorizerAttributeGetter,
+		authorizer:                authorizer,
+	}
+}
+func (ka *KubeletAuth) AuthenticateRequest(req *http.Request) (user.Info, bool, error) {
+	if ka.authenticator == nil {
+		return &user.DefaultInfo{}, true, nil
```

Comment:
Yeah, I guess that wouldn't make much sense, other than if there's a bug in the code or config which incorrectly admits unauthenticated users.